File: src/main/java/com/tambapps/http/restclient/util/ISSupplier.java
package com.tambapps.http.restclient.util;

import java.io.IOException;
import java.io.InputStream;

/**
 * Util class to supply an input stream
 */
public interface ISSupplier {

  /**
   * get the input stream
   * @return the input stream
   * @throws IOException in case of I/O error
   */
  InputStream get() throws IOException;

}

File: src/main/kotlin/com/tambapps/http/client/util/ISSupplier.kt
package com.tambapps.http.client.util

import java.io.IOException
import java.io.InputStream

/**
 * Util class to supply an input stream
 */
interface ISSupplier {
    /**
     * get the input stream
     * @return the input stream
     * @throws IOException in case of I/O error
     */
    @Throws(IOException::class)
    fun get(): InputStream
}